### DISCUSSION TITLE
Running the IOTA.c library on ESP micro controllers

### DISCUSSION THREAD

**Discussion by huggre** [iotaledger/iota.c#210] [Q&A] [answered]
Hi, would it be possible to run the IOTA.c library on microcontrollers such as the ESP8266 or ESP32?
If so, are there any tutorials as to how you compile and install the library on the controllers.
I'm working on a use case where I need to check the balance of a given IOTA address using an ESP based controller.
I'm familiar with the Arduino IDE and using existing Arduino libraries, but I'm not very familiar with C and C compilers in general.

**Comment by matjazv** [comment, marked as answer]
Hi huggre,
We have a separate repository for ESP32 microcontroller for iota.c library. You may find it here [https://github.com/iotaledger/esp32-client-sdk](https://github.com/iotaledger/esp32-client-sdk).
You will find a configuration and build instructions there.
If anything will still be unclear please let us know.

**Comment by oopsmonk** [comment]
@huggre first of all esp32 controllers doesn't have a dedicated IDE, it provides development frameworks such as [esp-idf](https://github.com/espressif/esp-idf), [esp-adf](https://github.com/espressif/esp-adf), [esp-mdf ](https://github.com/espressif/esp-mdf), [arduino-esp32](https://github.com/espressif/arduino-esp32) for different propose. 
as Matjaz mentioned, we provide [esp32-client-sdk](https://github.com/iotaledger/esp32-client-sdk) that it's an IOTA software development kit based on esp-idf for reducing development efforts, you might like to read [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/get-started/index.html#introduction) before dive into the SDK.

**Comment by huggre** [comment]
Thanks for your response guys
Was not aware of the ESP32 SDK so that may come in handy at some point.
However, for this particular use case I decided to go with a simple HTTP request to the IOTA node REST API.
Now I can continue using the Arduino IDE with any MCU that supports HTTP Requests
